cpp/src/lists/segmented_sort.cu

```diff
@@ -0,0 +1,246 @@
+    null_order null_precedence,
+    rmm::cuda_stream_view stream,
+    rmm::mr::device_memory_resource* mr)
+  {
+    CUDF_FAIL("segmented sort is not supported for non-numeric list types");
+  }
+  template <typename T>
+  std::enable_if_t<is_numeric<T>(), std::unique_ptr<column>> operator()(
+    column_view const& child,
+    column_view const& offsets,
+    order column_order,
+    null_order null_precedence,
+    rmm::cuda_stream_view stream,
+    rmm::mr::device_memory_resource* mr)
+  {
+    auto output =
+      cudf::detail::allocate_like(child, child.size(), mask_allocation_policy::NEVER, stream, mr);
+    mutable_column_view mutable_output_view = output->mutable_view();
+
+    auto keys = [&]() {
+      if (child.nullable()) {
+        rmm::device_uvector<T> keys(child.size(), stream);
+        auto null_replace_T = null_precedence == null_order::AFTER ? std::numeric_limits<T>::max()
+                                                                   : std::numeric_limits<T>::min();
+        auto device_child = column_device_view::create(child, stream);
+        auto keys_in =
+          cudf::detail::make_null_replacement_iterator<T>(*device_child, null_replace_T);
+        thrust::copy_n(rmm::exec_policy(stream), keys_in, child.size(), keys.begin());
+        return keys;
+      }
+      return rmm::device_uvector<T>{0, stream};
+    }();
+
+    std::unique_ptr<column> sorted_indices = cudf::make_numeric_column(
+      data_type(type_to_id<size_type>()), child.size(), mask_state::UNALLOCATED, stream, mr);
+    mutable_column_view mutable_indices_view = sorted_indices->mutable_view();
+    thrust::sequence(rmm::exec_policy(stream),
+                     mutable_indices_view.begin<size_type>(),
+                     mutable_indices_view.end<size_type>(),
+                     0);
+
+    if (column_order == order::ASCENDING)
+      SortPairsAscending(child.nullable() ? keys.data() : child.begin<T>(),
+                         mutable_output_view.begin<T>(),
+                         mutable_indices_view.begin<size_type>(),
+                         mutable_indices_view.begin<size_type>(),
+                         child.size(),
+                         offsets.size() - 1,
+                         offsets.begin<size_type>(),
+                         offsets.begin<size_type>() + 1,
+                         stream);
+    else
+      SortPairsDescending(child.nullable() ? keys.data() : child.begin<T>(),
+                          mutable_output_view.begin<T>(),
+                          mutable_indices_view.begin<size_type>(),
+                          mutable_indices_view.begin<size_type>(),
+                          child.size(),
+                          offsets.size() - 1,
+                          offsets.begin<size_type>(),
+                          offsets.begin<size_type>() + 1,
+                          stream);
+    std::vector<std::unique_ptr<column>> output_cols;
+    output_cols.push_back(std::move(output));
```

Comment:
Is this possible?
```suggestion
    std::vector<std::unique_ptr<column>> output_cols{std::move(output)};
```

Reply:
This was resolved but with no comment. So not possible?

Reply:
I committed it via github web. (which automatically resolves this) then it didn't compile. (reason: deleted function. Can't move initializer list of unique_ptr). I rolled back this change.